### PR TITLE
[terraform] Update terraform to v0.12.29

### PR DESCRIPTION
**Pull request by xSAVIKx**
In this PR I have updated the Terraform version to the latest v0.12.29.

### CODE REVIEW

**Comment by xSAVIKx** [comment]
@LOZORD @prabenzom, I've created a new PR instead of the #395. PTAL

**Comment by LOZORD** [comment]
Thanks! I'll take a look at this tomorrow!

File: terraform/README.markdown [hunk 1]
```diff
@@ -1,50 +1,96 @@
...
-Terraform stores state information about infrastructure it has provisioned. It uses this to plan out the delta between what your .tf files specifiy, and what's actually out there. This state can be stored in different ways by Terraform; it is configured via [backends](https://www.terraform.io/docs/backends/).
 
-The default backend for Terraform is local, which will store state information the working directory in ```$ ./.terraform```. Most build platforms (including GCE) do not persist the working directory between builds. Losing this state information is no bueno.
+Terraform stores state information about infrastructure it has provisioned. 
+It uses this to plan out the delta between what your .tf files specifiy, and what's actually out there. 
+This state can be stored in different ways by Terraform; it is configured via 
+[backends][terraform-backends].
+
+The default backend for Terraform is local, which will store state information 
+the working directory in `$ ./.terraform`. Most build platforms (including GCE)
```
Comment (LOZORD) [inline terraform/README.markdown:37]: Nit: since this isn't a command, remove the `$`?

File: terraform/README.markdown [hunk 2]
```diff
@@ -1,50 +1,96 @@
...
 
 There are a couple of options for managing Terraform state across builds:
 
+[terraform-backends]: https://www.terraform.io/docs/backends/
+
 ###### Ignore the issue
-In your build, you'll want to initialize terraform and refresh the local state. This is really not a good idea; it'll be slow and not multi-run safe (if multiple runs kick off concurrently, there'll be nastiness such as race conditions). [local_backend](examples/local_backend/README.markdown) is an example of this approach.
+
+In your build, you'll want to initialize terraform and refresh the local state. 
+This is really not a good idea; it'll be slow and not multi-run safe
```
Comment (LOZORD) [inline terraform/README.markdown:47]: Suggestion: __bold__ this so people see it?

File: terraform/README.markdown (hunk 1 above)
Comment (xSAVIKx) [reply terraform/README.markdown]: Dropped `$`.

File: terraform/README.markdown (hunk 2 above)
Comment (xSAVIKx) [reply terraform/README.markdown]: Done.

**Comment by xSAVIKx** [comment]
@LOZORD Fixed the comments. As I don't have the `write` access, could you please merge the PR?

**Pull request merged**